ts/IdentificationModal.tsx

```diff
@@ -128,18 +139,117 @@ class IdentificationModal extends React.PureComponent<Props, State> {
     };
   }
 
+  public componentDidMount() {
+    AppState.addEventListener("change", this.appStateChangeListener.bind(this));
+  }
+
+  public componentDidUnMount() {
+    AppState.removeEventListener(
+      "change",
+      this.appStateChangeListener.bind(this)
+    );
+  }
+
   public componentWillMount() {
-    TouchID.isSupported(isSupportedConfig)
-      .then(
-        biometryType => (biometryType === true ? "Fingerprint" : biometryType),
-        _ => undefined
-      )
-      .then(biometryType => this.setState({ biometryType }), _ => 0);
+    const { isFingerprintEnabled } = this.props;
+
+    if (isFingerprintEnabled) {
+      TouchID.isSupported(isSupportedConfig)
+        .then(
+          biometryType =>
+            biometryType === true ? "Fingerprint" : biometryType,
+          _ => undefined
+        )
+        .then(biometryType => this.setState({ biometryType }), _ => 0);
+    }
   }
 
-  public render() {
+  /**
+   * Check if fingerprint login can be prompted by looking at two parameters in
+   * serie:
+   * 1. `isFingerprintEnabled` coming from global state persisted preferences
+   * 2. The current status of biometry recognition support provided by querying
+   * the library in charge.
+   *
+   * It gets `updateBiometrySupportProp` as a boolean because it can be run
+   * from several contexts. When getting called while the app is returning
+   * foreground from background, biometry support has to be updated if system
+   * preferences changes happened.
+   *
+   * @param updateBiometrySupportProp
+   */
+  private maybeTriggerFingerprintRequest(
```

Comment:
this function is called from two different places, and the function has some logic that is executed in both cases and some other logic that is executed only when `updateBiometrySupportPropType ` is true, and that param is true only in one of those two cases - so perhaps you need two functions: one implementing the common logic and another one implementing the extra logic - in one place you'll call just the former and in the other place you'll call both

Reply:
I've just made some edits. If it's not enought I was thinking of separating logics in two parts: `isFingerprintAvailable` and `isFingerprintAvailableAndUpdateState`. They both return a Promise that is used to trigger the fingerprint modal.